src/WebJobs.Script/Description/FunctionInvokerBase.cs

```diff
@@ -110,77 +86,31 @@ protected bool InitializeFileWatcherIfEnabled()
 
         public async Task Invoke(object[] parameters)
         {
+            FunctionInvocationContext context = GetContextFromParameters(parameters, Metadata);
+            await InvokeCore(parameters, context);
+        }
+
+        private static FunctionInvocationContext GetContextFromParameters(object[] parameters, FunctionMetadata metadata)
+        {
             // We require the ExecutionContext, so this will throw if one is not found.
             ExecutionContext functionExecutionContext = parameters.OfType<ExecutionContext>().First();
-            PopulateExecutionContext(functionExecutionContext);
+            functionExecutionContext.FunctionDirectory = metadata.FunctionDirectory;
+            functionExecutionContext.FunctionName = metadata.Name;
 
             // These may not be present, so null is okay.
             TraceWriter functionTraceWriter = parameters.OfType<TraceWriter>().FirstOrDefault();
             Binder binder = parameters.OfType<Binder>().FirstOrDefault();
             ILogger logger = parameters.OfType<ILogger>().FirstOrDefault();
-            string invocationId = functionExecutionContext.InvocationId.ToString();
-
-            var startedEvent = new FunctionStartedEvent(functionExecutionContext.InvocationId, Metadata);
-            _metrics.BeginEvent(startedEvent);
-            var invokeLatencyEvent = LogInvocationMetrics(_metrics, Metadata);
-            var invocationStopWatch = new Stopwatch();
-            invocationStopWatch.Start();
-
-            try
-            {
-                string startMessage = $"Function started (Id={invocationId})";
-                TraceWriter.Info(startMessage);
-                Logger?.LogInformation(startMessage);
-
-                FunctionInvocationContext context = new FunctionInvocationContext
-                {
-                    ExecutionContext = functionExecutionContext,
-                    Binder = binder,
-                    TraceWriter = functionTraceWriter,
-                    Logger = logger
-                };
-
-                await InvokeCore(parameters, context);
 
-                invocationStopWatch.Stop();
-                LogFunctionResult(startedEvent, true, invocationId, invocationStopWatch.ElapsedMilliseconds);
-            }
-            catch (AggregateException ex)
+            FunctionInvocationContext context = new FunctionInvocationContext
             {
-                ExceptionDispatchInfo exInfo = null;
-
-                // If there's only a single exception, rethrow it by itself
-                Exception singleEx = ex.Flatten().InnerExceptions.SingleOrDefault();
```

Comment:
Which code? This looked like it was just standard catch & rethrow goo. What functionality specifically are we losing? 


---
In reply to: [126828161](https://github.com/Azure/azure-webjobs-sdk-script/pull/1660#discussion_r126828161) [](ancestors = 126828161)